### PR TITLE
vim-patch: update runtime files

### PR DESCRIPTION
**Pull request by clason**
- vim-patch:f5356bf6751f
- vim-patch:fc93594d562d
- vim-patch:8b2457a38198

### CODE REVIEW

**Review by zeertzjq — COMMENTED** [review]
Missing `runtime/autoload/cargo.vim`

File: runtime/doc/ft_rust.txt [hunk 1]
```diff
@@ -191,49 +428,52 @@ COMMANDS						       *rust-commands*
...
-==============================================================================
-MAPPINGS						       *rust-mappings*
+		When the current buffer is outside cargo project, the command
+		runs `rustc --test` command instead of "cargo test" as
+		fallback. All tests are run regardless of adding ! since there
+		is no way to run specific test function with rustc. [options]
+		is passed to `rustc` command arguments in the case.
 
-This plugin defines mappings for |[[| and |]]| to support hanging indents.
+		Takes optional modifiers (see |<mods>|):  >
```
Comment (seandewar) [inline runtime/doc/ft_rust.txt:455]:
```suggestion
		Takes optional modifiers (see |<mods>|):  >vim
```

File: runtime/doc/ft_rust.txt [hunk 2]
```diff
@@ -77,62 +79,297 @@ g:rust_fold~
...
+
+Invoking Cargo ~
+
+This plug defines very simple shortcuts for invoking Cargo from with Vim.
+
+:Cargo <args>                                                       *:Cargo*
+                Runs `cargo` with the provided arguments.
+
+:Cbuild <args>                                                     *:Cbuild*
+                Shortcut for `cargo build`.
```
[the remark this answers is not in the document]
Comment (clason) [reply runtime/doc/ft_rust.txt:292]: I only target the vimdoc parser :>

[the remark this answers is not in the document]
Comment (seandewar) [reply runtime/doc/ft_rust.txt:292]: In that case, carry on :innocent:

**Review by seandewar — APPROVED** [review]

**Pull request merged**